apiserver/facades/client/applicationoffers/base.go

```diff
@@ -34,50 +37,48 @@ type BaseAPI struct {
 	controllerUUID            string
-	modelUUID                 model.UUID
+	modelService              ModelService
 }
 
 // checkAdmin ensures that the specified in user is a model or controller admin.
 func (api *BaseAPI) checkAdmin(
-	ctx context.Context, user names.UserTag, modelID model.UUID, backend Backend,
+	ctx context.Context, user names.UserTag, modelUUID model.UUID,
 ) error {
-	err := api.Authorizer.EntityHasPermission(ctx, user, permission.SuperuserAccess, backend.ControllerTag())
+	controllerTag := names.NewControllerTag(api.controllerUUID)
+	err := api.Authorizer.EntityHasPermission(ctx, user, permission.SuperuserAccess, controllerTag)
 	if err != nil && !errors.Is(err, authentication.ErrorEntityMissingPermission) {
 		return errors.Trace(err)
 	} else if err == nil {
 		return nil
 	}
 
-	return api.Authorizer.EntityHasPermission(ctx, user, permission.AdminAccess, names.NewModelTag(modelID.String()))
+	return api.Authorizer.EntityHasPermission(ctx, user, permission.AdminAccess, names.NewModelTag(modelUUID.String()))
 }
 
 // checkControllerAdmin ensures that the logged in user is a controller admin.
 func (api *BaseAPI) checkControllerAdmin(ctx context.Context) error {
-	return api.Authorizer.HasPermission(ctx, permission.SuperuserAccess, api.ControllerModel.ControllerTag())
+	controllerTag := names.NewControllerTag(api.controllerUUID)
+	return api.Authorizer.HasPermission(ctx, permission.SuperuserAccess, controllerTag)
 }
 
-// modelForName looks up the model details for the named model and returns
-// the model (if found), the absolute model path which was used in the lookup,
-// and a bool indicating if the model was found,
-func (api *BaseAPI) modelForName(modelName, ownerName string) (Model, string, bool, error) {
+// modelForName returns the model details for the specified model name,
+// along with the absolute model path used in the lookup.
+//
+// The following errors may be returned:
+// - [coreerrors.NotFound] when no model with the given name exists.
+// - [coreerrors.NotValid] when ownerName is not valid.
+func (api *BaseAPI) modelForName(ctx context.Context, modelName, ownerName string) (model.Model, error) {
 	modelPath := fmt.Sprintf("%s/%s", ownerName, modelName)
-	var model Model
-	uuids, err := api.ControllerModel.AllModelUUIDs()
+	ownerUsername, err := coreuser.NewName(ownerName)
 	if err != nil {
-		return nil, modelPath, false, errors.Trace(err)
+		return model.Model{}, errors.Trace(err)
 	}
-	for _, uuid := range uuids {
-		m, release, err := api.StatePool.GetModel(uuid)
-		if err != nil {
-			return nil, modelPath, false, errors.Trace(err)
-		}
-		defer release()
-		if m.Name() == modelName && m.Owner().Id() == ownerName {
-			model = m
-			break
-		}
+	m, err := api.modelService.GetModelByNameAndOwner(ctx, modelName, ownerUsername)
+	if interrors.Is(err, modelerrors.NotFound) {
+		return model.Model{}, interrors.Errorf("model %q not found", modelPath).Add(coreerrors.NotFound)
```

Comment:
You mentioned it returns the `coreerrors.NotValid`:
modelerrors.UserNameNotValid -> coreerrors.NotValid

Reply:
Oh yes thanks for spotting this Kelvin